server/index.js

```diff
@@ -14,14 +39,44 @@ await fastify.register(cors, {
     // put your options here
 })
 
+async function craftNewWordFromCache(firstWord, secondWord) {
+    let cachedResult = await db.get('SELECT result, emoji FROM word_cache WHERE first_word = ? AND second_word = ?', [firstWord, secondWord]);
+
+    if (cachedResult) {
+        return cachedResult;
+    }
+
+    cachedResult = await db.get('SELECT result, emoji FROM word_cache WHERE first_word = ? AND second_word = ?', [secondWord, firstWord]);
+
+    return cachedResult;
+}
+
+async function cacheNewWord(firstWord, secondWord, result, emoji) {
+    await db.run('INSERT INTO word_cache (first_word, second_word, result, emoji) VALUES (?, ?, ?, ?)', [firstWord, secondWord, result, emoji]);
+}
+
 async function craftNewWord(firstWord, secondWord) {
+    const cachedResult = await craftNewWordFromCache(firstWord, secondWord);
+    if (cachedResult) {
+        return cachedResult;
+    }
+
+    if (!generate) {
```

Comment:
i don't think this code can be reached unless i missed where `generated` is mutated, also i do not think the client could handle that. if {result: '', emoji: ''} the client will just keep one of the original resources.

Reply:
You're right, generate was a constant.